atom/browser/atom_autofill_driver_factory.cc

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/atom_autofill_driver_factory.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "atom/browser/atom_autofill_driver.h"
+#include "base/bind.h"
+#include "base/callback.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+
+namespace atom {
+
+namespace {
+
+std::unique_ptr<AutofillDriver> CreateDriver(
+    content::RenderFrameHost* render_frame_host) {
+  return std::make_unique<AutofillDriver>(render_frame_host);
+}
+
+}  // namespace
+
+const char
+    AutofillDriverFactory::kAtomAutofillDriverFactoryWebContentsUserDataKey[] =
+        "atom_web_contents_autofill_driver_factory";
+
+AutofillDriverFactory::~AutofillDriverFactory() {}
+
+// static
+void AutofillDriverFactory::CreateForWebContents(
+    content::WebContents* contents) {
+  if (FromWebContents(contents))
+    return;
+
+  auto new_factory = std::make_unique<AutofillDriverFactory>(contents);
+  const std::vector<content::RenderFrameHost*> frames =
+      contents->GetAllFrames();
+  for (content::RenderFrameHost* frame : frames) {
+    if (frame->IsRenderFrameLive())
+      new_factory->RenderFrameCreated(frame);

```

Comment:
This can be moved to constructor of `AutofillDriverFactory`. There is no need to implement `AutofillDriverFactory::CreateForWebContents` and `FromWebContents` after making the class inherit `content::WebContentsUserData`.